### PR TITLE
Refactor Docs

### PR DESCRIPTION
**Pull request by fabioberger**
This PR:
Refactors the documentation to remove the reliance on DocType for documentation pages so that these components can be reused for Solidity documentation.

### CODE REVIEW

File: ts/utils/typedoc_utils.ts [hunk 1]
```diff
@@ -80,9 +82,24 @@ export const typeDocUtils = {
                 const allMembers = mainModuleExport.children;
                 const allMethods = _.filter(allMembers, typeDocUtils.isMethod);
                 const publicMethods = _.filter(allMethods, method => method.flags.isPublic);
-                menuSubsectionsBySection[menuItemName] = publicMethods;
+                const methodNames = _.map(publicMethods, t => t.name);
```
Comment (LogvinovLeon) [inline ts/utils/typedoc_utils.ts]: Rename `t` to smth useful like: `publicMethod`

File: ts/utils/typedoc_utils.ts [hunk 2]
```diff
@@ -68,7 +69,8 @@ export const typeDocUtils = {
                 const publicTypes = _.filter(allTypes, type => {
                     return typeDocUtils.isPublicType(type.name);
                 });
-                menuSubsectionsBySection[menuItemName] = publicTypes;
+                const typeNames = _.map(publicTypes, t => t.name);
```
Comment (LogvinovLeon) [inline ts/utils/typedoc_utils.ts]: Rename `t` to smth useful like: `publicType`

File: ts/utils/typedoc_utils.ts [hunk 3]
```diff
@@ -80,9 +82,24 @@ export const typeDocUtils = {
...
                 const allMethods = _.filter(allMembers, typeDocUtils.isMethod);
                 const publicMethods = _.filter(allMethods, method => method.flags.isPublic);
-                menuSubsectionsBySection[menuItemName] = publicMethods;
+                const methodNames = _.map(publicMethods, t => t.name);
+                menuSubsectionsBySection[menuItemName] = methodNames;
             }
         });
         return menuSubsectionsBySection;
     },
+    getFinal0xjsMenu(selectedVersion: string) {
```
Comment (LogvinovLeon) [inline ts/utils/typedoc_utils.ts]: Specify return type

File: ts/utils/typedoc_utils.ts (hunk 2 above)
Comment (fabioberger) [reply ts/utils/typedoc_utils.ts]: I think this is the only time where having a one letter variable name is acceptable. When the variable is hyper-localized. Similar to using `i` in a for loop. We wouldn't force developers to write `index`.

**Pull request merged**